can/can_graph.py

```diff
@@ -0,0 +1,294 @@
+def graph_can_ctf():
+    # Load required components from plugins
+    sink = system_plugins['ctf'].sink_component_classes['fs']
+    source = user_plugins['can'].source_component_classes['CANSource']
+
+    # Create graph and add components
+    graph = bt2.Graph()
+
+    graph_sink = graph.add_component(sink, 'test_sink',
+        params=bt2.MapValue({
+            'path': './ctf-full'
+        })
+    )
+
+    graph_source = graph.add_component(source, 'test_source',
+        params=bt2.MapValue({
+            'inputs': bt2.ArrayValue([CANSource_data_path]),
+            'databases': bt2.ArrayValue([CANSource_dbc_path])
+        })
+    )
+
+    # Connect components together
+    graph.connect_ports(
+        list(graph_source.output_ports.values())[0],
+        list(graph_sink.input_ports.values())[0]
+    )
+
+    # Run graph
+    graph.run()
+
+# BT2 graph - trims CANSource
+#
+# With the repo version of bt_plugin_gui, this will fail with
+#   Cannot make upstream message iterator initially seek
+#
+# You need to provide an implementation of
+#   def _user_seek_ns_from_origin(self, ns_from_origin):
+# in bt2._UserMessageIterator in order to make it work
+#
+def graph_can_filter_ctf():
+
+    # Load required components from plugins
+    sink = system_plugins['ctf'].sink_component_classes['fs']
+    filter = system_plugins['utils'].filter_component_classes['trimmer']
+    source = user_plugins['can'].source_component_classes['CANSource']
+
+    # Create graph and add components
+    graph = bt2.Graph()
+
+    graph_sink = graph.add_component(sink, 'test_sink',
+        params=bt2.MapValue({
+            'path' : './ctf-filtered'
+        })
+    )
+    graph_filter = graph.add_component(filter, 'test_filter',
+        params=bt2.MapValue({
+            'begin' : '0.000',
+            'end' : '1000.000'
+        })
+    )
+    graph_source = graph.add_component(source, 'test_source',
+        params=bt2.MapValue({
+            'inputs' : bt2.ArrayValue([CANSource_data_path]),
+            'databases' : bt2.ArrayValue([CANSource_dbc_path])
+        })
+    )
+
+    # Connect components together
+    graph.connect_ports(
+        list(graph_source.output_ports.values())[0],
+        list(graph_filter.input_ports.values())[0]
+    )
+    graph.connect_ports(
+        list(graph_filter.output_ports.values())[0],
+        list(graph_sink.input_ports.values())[0]
+    )
+
+    # Run graph
+    graph.run()
+
+# BT2 graph - trims existing CTF trace
+#
+# With the repo version of bt_plugin_gui, the utils.trimmer will fail with
+#   Babeltrace 2 library postcondition not satisfied;
+#   Packet message has no default clock snapshot:
+#
+# This seems to be a limitation of the 2.0.0. version of utils.trimmer filter.
+# You can work around it by extending the CANSource to generate a packet-enabled stream.
+#
+def graph_ctf_filter_ctf():
+
+    # Load required components from plugins
+    sink = system_plugins['ctf'].sink_component_classes['fs']
```

Comment:
Can you name these variables `ctf_fs_sink_comp_cls`, `trimmer_filter_comp_cls` and `ctf_fs_source_comp_cls`.  This is just to make sure to show that these are component *classes*.